components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts

```diff
@@ -31,17 +31,17 @@ export class HttpClientAuthInterceptor implements HttpInterceptor {
   }
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    // TODO(sr): Can we check if the request is sent to some external API?
+    //           Right now, we're throwing the ID token across the internet for telemetry,
+    //           where it's not needed. It would be nice to _not_ do that.
     let headers = request.headers.set('Authorization', `Bearer ${this.chefSession.id_token}`);
     const filtered = request.params.get('unfiltered') !== 'true';
-    // Uncomment here and after to clone() arg list
-    // after we've upgraded to angular 7.2+ (for this issue:
-    // https://github.com/angular/angular/issues/18812).
-    // const params = request.params.delete('unfiltered');
+    const params = request.params.delete('unfiltered');
     if (this.projects && filtered) {
       headers = headers.set('projects', this.projects);
     }
     return next
-      .handle(request.clone({ headers })).pipe(
+      .handle(request.clone({ headers, params })).pipe(
```

Comment:
Do you have to pass the `params` because they are updated? Where before they were part of the `request` object and were copied in the clone?

Reply:
❓I'll just admit it, I have no idea. Monkey see, monkey do. This could be _not the right thing_, I'm afraid. What do you think?

Reply:
That was my guess. That parameter is called "update" https://angular.io/api/common/http/HttpRequest.

Reply:
Thanks for the docs. I think that's what happens, yeah. :bulb:

Reply:
To answer your original question @lancewf : _yes_.
Here is what I observed examining live code:
The `clone` operation clones everything: body, header, params, and others. If you examine the data structures in live code, when we effectively copy the headers (`request.headers.set`) or the params (`request.params.delete`), it is not just a simple object with those fields. It sets the `update` field you noticed, to the delta, essentially instructions on what to change in the object when that is passed to the `clone` method.